io.quarkus...BootstrapMavenException resolving a wrong dependency version

### Describe the bug

It seems `io.quarkus.bootstrap.resolver.BootstrapAppModelResolver` is somehow resolving a wrong version for a transitive dependency in a quarkus based application.

I have a reproducer where `mvn dependency:tree` gives:

```
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ demo20221103xmlpullver ---
[INFO] org.acme:demo20221103xmlpullver:jar:1.0.0-SNAPSHOT
[INFO] +- io.quarkus:quarkus-arc:jar:2.13.3.Final:compile
[INFO] |  +- io.quarkus.arc:arc:jar:2.13.3.Final:compile
[INFO] |  |  +- jakarta.enterprise:jakarta.enterprise.cdi-api:jar:2.0.2:compile
[INFO] |  |  |  +- jakarta.el:jakarta.el-api:jar:3.0.3:compile
[INFO] |  |  |  \- jakarta.interceptor:jakarta.interceptor-api:jar:1.2.5:compile
[INFO] |  |  +- jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile
[INFO] |  |  +- jakarta.transaction:jakarta.transaction-api:jar:1.3.3:compile
[INFO] |  |  +- io.smallrye.reactive:mutiny:jar:1.7.0:compile
[INFO] |  |  |  +- org.reactivestreams:reactive-streams:jar:1.0.3:compile
[INFO] |  |  |  \- io.smallrye.common:smallrye-common-annotation:jar:1.13.1:compile
[INFO] |  |  \- org.jboss.logging:jboss-logging:jar:3.5.0.Final:compile
[INFO] |  +- io.quarkus:quarkus-core:jar:2.13.3.Final:compile
[INFO] |  |  +- jakarta.inject:jakarta.inject-api:jar:1.0:compile
[INFO] |  |  +- io.quarkus:quarkus-ide-launcher:jar:2.13.3.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-development-mode-spi:jar:2.13.3.Final:compile
[INFO] |  |  +- io.smallrye.config:smallrye-config:jar:2.12.0:compile
[INFO] |  |  |  \- io.smallrye.config:smallrye-config-core:jar:2.12.0:compile
[INFO] |  |  |     +- org.eclipse.microprofile.config:microprofile-config-api:jar:2.0.1:compile
[INFO] |  |  |     +- io.smallrye.common:smallrye-common-expression:jar:1.13.1:compile
[INFO] |  |  |     |  \- io.smallrye.common:smallrye-common-function:jar:1.13.1:compile
[INFO] |  |  |     +- io.smallrye.common:smallrye-common-classloader:jar:1.13.1:compile
[INFO] |  |  |     \- io.smallrye.config:smallrye-config-common:jar:2.12.0:compile
[INFO] |  |  +- org.jboss.logmanager:jboss-logmanager-embedded:jar:1.0.10:compile
[INFO] |  |  +- org.jboss.logging:jboss-logging-annotations:jar:2.2.1.Final:compile
[INFO] |  |  +- org.jboss.threads:jboss-threads:jar:3.4.3.Final:compile
[INFO] |  |  +- org.jboss.slf4j:slf4j-jboss-logmanager:jar:1.2.0.Final:compile
[INFO] |  |  +- org.graalvm.sdk:graal-sdk:jar:22.2.0:compile
[INFO] |  |  +- org.wildfly.common:wildfly-common:jar:1.5.4.Final-format-001:compile
[INFO] |  |  +- io.quarkus:quarkus-bootstrap-runner:jar:2.13.3.Final:compile
[INFO] |  |  \- io.quarkus:quarkus-fs-util:jar:0.0.9:compile
[INFO] |  \- org.eclipse.microprofile.context-propagation:microprofile-context-propagation-api:jar:1.2:compile
[INFO] +- io.quarkus:quarkus-resteasy-reactive:jar:2.13.3.Final:compile
[INFO] |  +- io.quarkus:quarkus-resteasy-reactive-common:jar:2.13.3.Final:compile
[INFO] |  |  +- io.quarkus.resteasy.reactive:resteasy-reactive-common:jar:2.13.3.Final:compile
[INFO] |  |  |  \- io.quarkus.resteasy.reactive:resteasy-reactive-common-types:jar:2.13.3.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-mutiny:jar:2.13.3.Final:compile
[INFO] |  |  |  +- io.quarkus:quarkus-smallrye-context-propagation:jar:2.13.3.Final:compile
[INFO] |  |  |  |  \- io.smallrye:smallrye-context-propagation:jar:1.2.2:compile
[INFO] |  |  |  |     +- io.smallrye:smallrye-context-propagation-api:jar:1.2.2:compile
[INFO] |  |  |  |     \- io.smallrye:smallrye-context-propagation-storage:jar:1.2.2:compile
[INFO] |  |  |  \- io.smallrye.reactive:mutiny-smallrye-context-propagation:jar:1.7.0:compile
[INFO] |  |  \- io.quarkus:quarkus-vertx:jar:2.13.3.Final:compile
[INFO] |  |     +- io.quarkus:quarkus-netty:jar:2.13.3.Final:compile
[INFO] |  |     |  +- io.netty:netty-codec:jar:4.1.82.Final:compile
[INFO] |  |     |  +- io.netty:netty-codec-http:jar:4.1.82.Final:compile
[INFO] |  |     |  +- io.netty:netty-codec-http2:jar:4.1.82.Final:compile
[INFO] |  |     |  +- io.netty:netty-handler:jar:4.1.82.Final:compile
[INFO] |  |     |  |  \- io.netty:netty-transport-native-unix-common:jar:4.1.82.Final:compile
[INFO] |  |     |  \- com.aayushatharva.brotli4j:brotli4j:jar:1.8.0:compile
[INFO] |  |     |     \- com.aayushatharva.brotli4j:native-osx-x86_64:jar:1.8.0:compile
[INFO] |  |     +- io.netty:netty-codec-haproxy:jar:4.1.82.Final:compile
[INFO] |  |     |  +- io.netty:netty-buffer:jar:4.1.82.Final:compile
[INFO] |  |     |  \- io.netty:netty-transport:jar:4.1.82.Final:compile
[INFO] |  |     +- io.quarkus:quarkus-vertx-latebound-mdc-provider:jar:2.13.3.Final:compile
[INFO] |  |     \- io.smallrye:smallrye-fault-tolerance-vertx:jar:5.5.0:compile
[INFO] |  +- io.quarkus.resteasy.reactive:resteasy-reactive-vertx:jar:2.13.3.Final:compile
[INFO] |  |  +- io.vertx:vertx-web:jar:4.3.4:compile
[INFO] |  |  |  +- io.vertx:vertx-web-common:jar:4.3.4:compile
[INFO] |  |  |  +- io.vertx:vertx-auth-common:jar:4.3.4:compile
[INFO] |  |  |  +- io.vertx:vertx-bridge-common:jar:4.3.4:compile
[INFO] |  |  |  \- io.vertx:vertx-core:jar:4.3.4:compile
[INFO] |  |  |     +- io.netty:netty-common:jar:4.1.82.Final:compile
[INFO] |  |  |     +- io.netty:netty-handler-proxy:jar:4.1.82.Final:compile
[INFO] |  |  |     |  \- io.netty:netty-codec-socks:jar:4.1.82.Final:compile
[INFO] |  |  |     +- io.netty:netty-resolver:jar:4.1.82.Final:compile
[INFO] |  |  |     +- io.netty:netty-resolver-dns:jar:4.1.82.Final:compile
[INFO] |  |  |     |  \- io.netty:netty-codec-dns:jar:4.1.82.Final:compile
[INFO] |  |  |     \- com.fasterxml.jackson.core:jackson-core:jar:2.13.4:compile
[INFO] |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-core:jar:2.27.0:compile
[INFO] |  |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-runtime:jar:2.27.0:compile
[INFO] |  |  |  \- io.smallrye.reactive:vertx-mutiny-generator:jar:2.27.0:compile
[INFO] |  |  |     \- io.vertx:vertx-codegen:jar:4.3.4:compile
[INFO] |  |  +- io.quarkus.resteasy.reactive:resteasy-reactive:jar:2.13.3.Final:compile
[INFO] |  |  +- org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.1_spec:jar:2.0.1.Final:compile
[INFO] |  |  +- org.jboss.logging:commons-logging-jboss-logging:jar:1.0.0.Final:compile
[INFO] |  |  \- org.jboss.spec.javax.xml.bind:jboss-jaxb-api_2.3_spec:jar:2.0.0.Final:compile
[INFO] |  +- io.quarkus:quarkus-vertx-http:jar:2.13.3.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-security-runtime-spi:jar:2.13.3.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-credentials:jar:2.13.3.Final:compile
[INFO] |  |  +- io.smallrye.common:smallrye-common-vertx-context:jar:1.13.1:compile
[INFO] |  |  |  \- io.smallrye.common:smallrye-common-constraint:jar:1.13.1:compile
[INFO] |  |  +- io.quarkus:quarkus-vertx-http-dev-console-runtime-spi:jar:2.13.3.Final:compile
[INFO] |  |  +- io.quarkus.security:quarkus-security:jar:1.1.4.Final:compile
[INFO] |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-web:jar:2.27.0:compile
[INFO] |  |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-web-common:jar:2.27.0:compile
[INFO] |  |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-auth-common:jar:2.27.0:compile
[INFO] |  |  |  +- io.smallrye.reactive:smallrye-mutiny-vertx-bridge-common:jar:2.27.0:compile
[INFO] |  |  |  \- io.smallrye.reactive:smallrye-mutiny-vertx-uri-template:jar:2.27.0:compile
[INFO] |  |  |     \- io.vertx:vertx-uri-template:jar:4.3.4:compile
[INFO] |  |  \- io.github.crac:org-crac:jar:0.1.1:compile
[INFO] |  \- io.quarkus:quarkus-jsonp:jar:2.13.3.Final:compile
[INFO] |     \- org.glassfish:jakarta.json:jar:1.1.6:compile
[INFO] +- org.drools:drools-drl-quarkus:jar:8.29.0.Final:compile
[INFO] |  +- org.drools:drools-engine:jar:8.29.0.Final:compile
[INFO] |  |  +- org.kie:kie-api:jar:8.29.0.Final:compile
[INFO] |  |  +- org.kie:kie-internal:jar:8.29.0.Final:compile
[INFO] |  |  +- org.drools:drools-core:jar:8.29.0.Final:compile
[INFO] |  |  |  +- org.kie:kie-util-xml:jar:8.29.0.Final:compile
[INFO] |  |  |  +- org.drools:drools-wiring-api:jar:8.29.0.Final:compile
[INFO] |  |  |  +- org.drools:drools-wiring-static:jar:8.29.0.Final:compile
[INFO] |  |  |  \- org.drools:drools-io:jar:8.29.0.Final:compile
[INFO] |  |  |     \- org.drools:drools-util:jar:8.29.0.Final:compile
[INFO] |  |  +- org.drools:drools-wiring-dynamic:jar:8.29.0.Final:compile
[INFO] |  |  +- org.drools:drools-kiesession:jar:8.29.0.Final:compile
[INFO] |  |  +- org.drools:drools-tms:jar:8.29.0.Final:compile
[INFO] |  |  +- org.drools:drools-compiler:jar:8.29.0.Final:compile
[INFO] |  |  |  +- org.drools:drools-drl-parser:jar:8.29.0.Final:compile
[INFO] |  |  |  +- org.drools:drools-drl-extensions:jar:8.29.0.Final:compile
[INFO] |  |  |  +- org.drools:drools-drl-ast:jar:8.29.0.Final:compile
[INFO] |  |  |  +- org.kie:kie-memory-compiler:jar:8.29.0.Final:compile
[INFO] |  |  |  +- org.drools:drools-ecj:jar:8.29.0.Final:compile
[INFO] |  |  |  +- org.kie:kie-util-maven-support:jar:8.29.0.Final:compile
[INFO] |  |  |  \- org.antlr:antlr-runtime:jar:3.5.2:compile
[INFO] |  |  +- org.drools:drools-model-compiler:jar:8.29.0.Final:compile
[INFO] |  |  \- org.drools:drools-model-codegen:jar:8.29.0.Final:compile
[INFO] |  |     +- org.drools:drools-codegen-common:jar:8.29.0.Final:compile
[INFO] |  |     +- com.github.javaparser:javaparser-core:jar:3.24.2:compile
[INFO] |  |     +- org.drools:drools-mvel-parser:jar:8.29.0.Final:compile
[INFO] |  |     \- org.drools:drools-mvel-compiler:jar:8.29.0.Final:compile
[INFO] |  \- io.quarkus.arc:arc-processor:jar:2.13.3.Final:compile
[INFO] |     +- org.jboss:jandex:jar:2.4.3.Final:compile
[INFO] |     \- io.quarkus.gizmo:gizmo:jar:1.1.1.Final:compile
[INFO] |        +- org.ow2.asm:asm:jar:9.3:compile
[INFO] |        \- org.ow2.asm:asm-util:jar:9.3:compile
[INFO] |           +- org.ow2.asm:asm-tree:jar:9.3:compile
[INFO] |           \- org.ow2.asm:asm-analysis:jar:9.3:compile
[INFO] +- org.drools:drools-ruleunits-engine:jar:8.29.0.Final:compile
[INFO] |  +- org.drools:drools-ruleunits-dsl:jar:8.29.0.Final:compile
[INFO] |  |  +- org.drools:drools-canonical-model:jar:8.29.0.Final:compile
[INFO] |  |  \- org.drools:drools-ruleunits-impl:jar:8.29.0.Final:compile
[INFO] |  |     \- org.drools:drools-ruleunits-api:jar:8.29.0.Final:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:1.7.36:compile
[INFO] +- org.assertj:assertj-core:jar:3.23.1:test
[INFO] |  \- net.bytebuddy:byte-buddy:jar:1.12.9:test
[INFO] +- io.quarkus:quarkus-junit5:jar:2.13.3.Final:test
[INFO] |  +- io.quarkus:quarkus-bootstrap-core:jar:2.13.3.Final:test
[INFO] |  |  +- io.quarkus:quarkus-bootstrap-app-model:jar:2.13.3.Final:test
[INFO] |  |  \- io.smallrye.common:smallrye-common-io:jar:1.13.1:compile
[INFO] |  +- org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.5:test
[INFO] |  +- io.quarkus:quarkus-test-common:jar:2.13.3.Final:test
[INFO] |  |  +- io.quarkus:quarkus-core-deployment:jar:2.13.3.Final:test
[INFO] |  |  |  +- org.aesh:readline:jar:2.2:test
[INFO] |  |  |  |  \- org.fusesource.jansi:jansi:jar:1.18:test
[INFO] |  |  |  +- org.aesh:aesh:jar:2.6:test
[INFO] |  |  |  +- org.ow2.asm:asm-commons:jar:9.3:test
[INFO] |  |  |  +- io.quarkus:quarkus-class-change-agent:jar:2.13.3.Final:test
[INFO] |  |  |  +- io.quarkus:quarkus-devtools-utilities:jar:2.13.3.Final:test
[INFO] |  |  |  +- io.quarkus:quarkus-builder:jar:2.13.3.Final:test
[INFO] |  |  |  \- org.junit.platform:junit-platform-launcher:jar:1.9.1:test
[INFO] |  |  +- io.quarkus:quarkus-bootstrap-maven-resolver:jar:2.13.3.Final:test
[INFO] |  |  |  +- org.apache.maven:maven-embedder:jar:3.8.6:test
[INFO] |  |  |  |  +- org.apache.maven:maven-settings:jar:3.8.6:test
[INFO] |  |  |  |  +- org.apache.maven:maven-core:jar:3.8.6:test
[INFO] |  |  |  |  |  +- org.apache.maven:maven-artifact:jar:3.8.6:test
[INFO] |  |  |  |  |  \- org.codehaus.plexus:plexus-component-annotations:jar:2.1.0:test
[INFO] |  |  |  |  +- org.apache.maven:maven-plugin-api:jar:3.8.6:test
[INFO] |  |  |  |  +- org.apache.maven:maven-model:jar:3.8.6:test
[INFO] |  |  |  |  +- org.apache.maven:maven-model-builder:jar:3.8.6:test
[INFO] |  |  |  |  +- org.apache.maven:maven-builder-support:jar:3.8.6:test
[INFO] |  |  |  |  +- org.apache.maven.resolver:maven-resolver-api:jar:1.6.3:test
[INFO] |  |  |  |  +- org.apache.maven.resolver:maven-resolver-util:jar:1.6.3:test
[INFO] |  |  |  |  +- org.apache.maven.shared:maven-shared-utils:jar:3.3.4:test
[INFO] |  |  |  |  |  \- commons-io:commons-io:jar:2.11.0:test
[INFO] |  |  |  |  +- com.google.inject:guice:jar:no_aop:4.2.2:test
[INFO] |  |  |  |  |  \- com.google.guava:guava:jar:31.1-jre:test
[INFO] |  |  |  |  |     \- com.google.guava:failureaccess:jar:1.0.1:test
[INFO] |  |  |  |  +- org.codehaus.plexus:plexus-classworlds:jar:2.6.0:test
[INFO] |  |  |  |  +- org.codehaus.plexus:plexus-sec-dispatcher:jar:2.0:test
[INFO] |  |  |  |  +- org.codehaus.plexus:plexus-cipher:jar:2.0:test
[INFO] |  |  |  |  \- commons-cli:commons-cli:jar:1.4:test
[INFO] |  |  |  +- org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.3.5:test
[INFO] |  |  |  +- org.apache.maven:maven-settings-builder:jar:3.8.6:test
[INFO] |  |  |  |  \- org.codehaus.plexus:plexus-interpolation:jar:1.26:test
[INFO] |  |  |  +- org.apache.maven:maven-resolver-provider:jar:3.8.6:test
[INFO] |  |  |  |  +- org.apache.maven:maven-repository-metadata:jar:3.8.6:test
[INFO] |  |  |  |  +- org.apache.maven.resolver:maven-resolver-spi:jar:1.6.3:test
[INFO] |  |  |  |  +- org.apache.maven.resolver:maven-resolver-impl:jar:1.6.3:test
[INFO] |  |  |  |  \- org.codehaus.plexus:plexus-utils:jar:3.3.0:test
[INFO] |  |  |  +- org.apache.maven.resolver:maven-resolver-connector-basic:jar:1.6.3:test
[INFO] |  |  |  +- org.apache.maven.resolver:maven-resolver-transport-wagon:jar:1.6.3:test
[INFO] |  |  |  +- org.apache.maven.wagon:wagon-http:jar:3.5.1:test
[INFO] |  |  |  |  +- org.apache.maven.wagon:wagon-http-shared:jar:3.5.1:test
[INFO] |  |  |  |  \- org.apache.maven.wagon:wagon-provider-api:jar:3.5.1:test
[INFO] |  |  |  \- org.apache.maven.wagon:wagon-file:jar:3.5.1:test
[INFO] |  |  \- io.quarkus:quarkus-bootstrap-gradle-resolver:jar:2.13.3.Final:test
[INFO] |  +- io.quarkus:quarkus-junit5-properties:jar:2.13.3.Final:test
[INFO] |  +- org.junit.jupiter:junit-jupiter:jar:5.9.1:test
[INFO] |  |  +- org.junit.jupiter:junit-jupiter-api:jar:5.9.1:test
[INFO] |  |  |  +- org.opentest4j:opentest4j:jar:1.2.0:test
[INFO] |  |  |  +- org.junit.platform:junit-platform-commons:jar:1.9.1:test
[INFO] |  |  |  \- org.apiguardian:apiguardian-api:jar:1.1.2:test
[INFO] |  |  +- org.junit.jupiter:junit-jupiter-params:jar:5.9.1:test
[INFO] |  |  \- org.junit.jupiter:junit-jupiter-engine:jar:5.9.1:test
[INFO] |  |     \- org.junit.platform:junit-platform-engine:jar:1.9.1:test
[INFO] |  \- com.thoughtworks.xstream:xstream:jar:1.4.19:test
[INFO] |     \- io.github.x-stream:mxparser:jar:1.2.2:test
[INFO] |        \- xmlpull:xmlpull:jar:1.1.3.1:test
[INFO] \- io.rest-assured:rest-assured:jar:4.5.1:test
[INFO]    +- org.codehaus.groovy:groovy:jar:3.0.9:test
[INFO]    +- org.codehaus.groovy:groovy-xml:jar:3.0.9:test
[INFO]    +- org.apache.httpcomponents:httpclient:jar:4.5.13:test
[INFO]    |  +- org.apache.httpcomponents:httpcore:jar:4.4.15:test
[INFO]    |  \- commons-codec:commons-codec:jar:1.15:compile
[INFO]    +- org.apache.httpcomponents:httpmime:jar:4.5.13:test
[INFO]    +- org.hamcrest:hamcrest:jar:2.1:test
[INFO]    +- org.ccil.cowan.tagsoup:tagsoup:jar:1.2.1:test
[INFO]    +- io.rest-assured:json-path:jar:4.5.1:test
[INFO]    |  +- org.codehaus.groovy:groovy-json:jar:3.0.9:test
[INFO]    |  \- io.rest-assured:rest-assured-common:jar:4.5.1:test
[INFO]    \- io.rest-assured:xml-path:jar:4.5.1:test
[INFO]       \- org.apache.commons:commons-lang3:jar:3.12.0:test
```

please notice the version of `xmlpull` is: `xmlpull:xmlpull:jar:1.1.3.1` accordingly to Maven.

### Expected behavior

- `mvn clean install` to finish correctly.

- `xmlpull:xmlpull:jar:1.1.3.1` is considered as indicated by Maven, not `1.2.0`.

### Actual behavior

```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.13.3.Final:generate-code (default) on project demo20221103xmlpullver: Quarkus code generation phase has failed: Failed to bootstrap application in NORMAL mode: Failed to resolve artifact xmlpull:xmlpull:jar:1.2.0: xmlpull:xmlpull:jar:1.2.0 was not found in https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.13.3.Final:generate-code (default) on project demo20221103xmlpullver: Quarkus code generation phase has failed
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:577)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Quarkus code generation phase has failed
    at io.quarkus.maven.GenerateCodeMojo.generateCode (GenerateCodeMojo.java:85)
    at io.quarkus.maven.GenerateCodeMojo.doExecute (GenerateCodeMojo.java:53)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute (QuarkusBootstrapMojo.java:154)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:577)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to bootstrap application in NORMAL mode
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.doBootstrap (QuarkusBootstrapProvider.java:243)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.bootstrapApplication (QuarkusBootstrapProvider.java:291)
    at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication (QuarkusBootstrapProvider.java:97)
    at io.quarkus.maven.QuarkusBootstrapMojo.bootstrapApplication (QuarkusBootstrapMojo.java:272)
    at io.quarkus.maven.GenerateCodeMojo.generateCode (GenerateCodeMojo.java:70)
    at io.quarkus.maven.GenerateCodeMojo.doExecute (GenerateCodeMojo.java:53)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute (QuarkusBootstrapMojo.java:154)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:577)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: io.quarkus.bootstrap.resolver.maven.BootstrapMavenException: Failed to resolve artifact xmlpull:xmlpull:jar:1.2.0
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveInternal (MavenArtifactResolver.java:170)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolve (MavenArtifactResolver.java:155)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visitLeave (BuildDependencyGraphVisitor.java:141)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:92)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:75)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:75)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:82)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:82)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:82)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyTreeResolver.resolve (ApplicationDependencyTreeResolver.java:210)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.buildAppModel (BootstrapAppModelResolver.java:321)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.doResolveModel (BootstrapAppModelResolver.java:286)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveManagedModel (BootstrapAppModelResolver.java:165)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.doBootstrap (QuarkusBootstrapProvider.java:240)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.bootstrapApplication (QuarkusBootstrapProvider.java:291)
    at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication (QuarkusBootstrapProvider.java:97)
    at io.quarkus.maven.QuarkusBootstrapMojo.bootstrapApplication (QuarkusBootstrapMojo.java:272)
    at io.quarkus.maven.GenerateCodeMojo.generateCode (GenerateCodeMojo.java:70)
    at io.quarkus.maven.GenerateCodeMojo.doExecute (GenerateCodeMojo.java:53)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute (QuarkusBootstrapMojo.java:154)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:577)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: xmlpull:xmlpull:jar:1.2.0 was not found in https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:425)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:229)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:207)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact (DefaultRepositorySystem.java:262)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveInternal (MavenArtifactResolver.java:165)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolve (MavenArtifactResolver.java:155)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visitLeave (BuildDependencyGraphVisitor.java:141)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:92)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:75)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:75)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:82)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:82)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:82)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyTreeResolver.resolve (ApplicationDependencyTreeResolver.java:210)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.buildAppModel (BootstrapAppModelResolver.java:321)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.doResolveModel (BootstrapAppModelResolver.java:286)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveManagedModel (BootstrapAppModelResolver.java:165)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.doBootstrap (QuarkusBootstrapProvider.java:240)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.bootstrapApplication (QuarkusBootstrapProvider.java:291)
    at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication (QuarkusBootstrapProvider.java:97)
    at io.quarkus.maven.QuarkusBootstrapMojo.bootstrapApplication (QuarkusBootstrapMojo.java:272)
    at io.quarkus.maven.GenerateCodeMojo.generateCode (GenerateCodeMojo.java:70)
    at io.quarkus.maven.GenerateCodeMojo.doExecute (GenerateCodeMojo.java:53)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute (QuarkusBootstrapMojo.java:154)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:577)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: xmlpull:xmlpull:jar:1.2.0 was not found in https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced
    at org.eclipse.aether.internal.impl.DefaultUpdateCheckManager.newException (DefaultUpdateCheckManager.java:217)
    at org.eclipse.aether.internal.impl.DefaultUpdateCheckManager.checkArtifact (DefaultUpdateCheckManager.java:189)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.gatherDownloads (DefaultArtifactResolver.java:574)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads (DefaultArtifactResolver.java:484)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:402)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:229)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:207)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact (DefaultRepositorySystem.java:262)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveInternal (MavenArtifactResolver.java:165)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolve (MavenArtifactResolver.java:155)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visitLeave (BuildDependencyGraphVisitor.java:141)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:92)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:75)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:75)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:82)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:82)
    at io.quarkus.bootstrap.resolver.maven.BuildDependencyGraphVisitor.visit (BuildDependencyGraphVisitor.java:82)
    at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyTreeResolver.resolve (ApplicationDependencyTreeResolver.java:210)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.buildAppModel (BootstrapAppModelResolver.java:321)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.doResolveModel (BootstrapAppModelResolver.java:286)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveManagedModel (BootstrapAppModelResolver.java:165)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.doBootstrap (QuarkusBootstrapProvider.java:240)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.bootstrapApplication (QuarkusBootstrapProvider.java:291)
    at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication (QuarkusBootstrapProvider.java:97)
    at io.quarkus.maven.QuarkusBootstrapMojo.bootstrapApplication (QuarkusBootstrapMojo.java:272)
    at io.quarkus.maven.GenerateCodeMojo.generateCode (GenerateCodeMojo.java:70)
    at io.quarkus.maven.GenerateCodeMojo.doExecute (GenerateCodeMojo.java:53)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute (QuarkusBootstrapMojo.java:154)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:577)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

### How to Reproduce?

1. checkout https://github.com/tarilabs/demo20221103xmlpullver
2. `mvn clean install -X`
3. it exhibits the problem, which seems to me Quarkus' `io.quarkus.bootstrap.resolver.BootstrapAppModelResolver` and related, is trying to resolve a dependency version (`1.2.0`) which is not the actual one in the current app, accordingly to Maven: `1.1.3.1`

Workaround exists, by adding:
```xml
<repositories>
    <repository>
      <id>central</id>
      <name>Central Repository</name>
      <url>https://repo.maven.apache.org/maven2</url>
      <layout>default</layout>
      <snapshots>
        <enabled>false</enabled>
      </snapshots>
    </repository>
    <repository>
      <id>jboss-public-repository-group</id>
      <name>JBoss Public Repository Group</name>
      <url>https://repository.jboss.org/nexus/content/groups/public/</url>
      <releases>
        <enabled>true</enabled>
        <updatePolicy>never</updatePolicy>
      </releases>
      <snapshots>
        <enabled>true</enabled>
        <updatePolicy>daily</updatePolicy>
      </snapshots>
    </repository>
  </repositories>
```

to the `pom.xml` of the Quarkus reproducer application, then it resolves the non-needed-dependency in other repository (JBoss) and then the build progresses and completes fine.

### Output of `uname -a` or `ver`

Darwin mmortari1-mac 20.6.0 Darwin Kernel Version 20.6.0

### Output of `java -version`

openjdk version "18.0.1.1" 2022-04-22

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.13.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

/cc @mariofusco @evacchi

--- a/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BuildDependencyGraphVisitor.java
+++ b/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BuildDependencyGraphVisitor.java
@@ -138,7 +138,7 @@ public class BuildDependencyGraphVisitor {
 
             Artifact artifact = dep.getArtifact();
             if (artifact.getFile() == null) {
-                artifact = resolver.resolve(artifact).getArtifact();
+                artifact = resolver.resolve(artifact, node.getRepositories()).getArtifact();
             }
 
             int flags = DependencyFlags.DEPLOYMENT_CP;

